core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -137,7 +138,7 @@ abstract class AbstractFetcherThread(name: String,
                     // Once we hand off the partition data to the subclass, we can't mess with it any more in this thread
                     processPartitionData(topicAndPartition, currentPartitionFetchState.offset, partitionData)
                   } catch {
-                    case ime: InvalidMessageException =>
+                    case ime: CorruptRecordException =>
```

Comment:
OK, that makes sense for that case. We also changed a number of places to throw CorruptRecordException instead of InvalidMessageException and that is what I am concerned about here. I am hoping that these cases do not affect old clients. Is that right?